flocker/volume/test/filesystemtests.py

```diff
@@ -618,7 +620,11 @@ def got_volumes(copied):
                     with from_filesystem.reader() as reader:
                         with to_filesystem.writer() as writer:
                             data = reader.read()
-                            writer.write(data[1:])
+                            try:
+                                writer.write(data[1:])
+                            except IOError as e:
+                                if e.errno != errno.EPIPE:
```

Comment:
Now that I've tried to formulate the suggested comment I am wondering why we write `data[1:]` here, that is, why we skip the first byte... That makes the stream invalid and, thus, a reader that works asynchronously may error out  after seeing the corrupted stream head and won't wait for the rest of the stream. This is what causes `EPIPE` if the writer is not fast enough.
From the test description it is not clear why we would want to write an invalid stream.
Perhaps the intention was to write `data[:-1]` to ensure that the reader does not get a complete stream before the exception happens? @itamarst ?